.github/workflows/kakarot_rpc.yml

```diff
@@ -1,59 +1,97 @@
+        with:
+          name: artifacts
+          path: ./artifacts
+  # Inspired by Reth CI.
+  # <https://github.com/paradigmxyz/reth/blob/027d50fc105fd527dca0bf56fe51b7240f119e66/.github/workflows/docker.yml>
   build:
     runs-on: ubuntu-latest-16-cores
+    needs: prepare
     strategy:
       matrix:
-        platform:
-          - linux/amd64
-          - linux/arm64
         include:
-          - platform: linux/amd64
-            STARKNET_BIN_DIR: dsrifc349xgnwsv661shqljxb8pbxx2j
-            SINK_BIN_DIR: vr6f0crii7571q19j122dfqb1ha1m4i9
-          - platform: linux/arm64
-            STARKNET_BIN_DIR: f7va4mjqww1kkpp4il6y295dgcwq147v
-            SINK_BIN_DIR: 3iqnrcirqpg4s7zdy1wdh0dq17jwzmlc
+          - platform: amd64
+            target: x86_64-unknown-linux-gnu
+          - platform: arm64
+            target: aarch64-unknown-linux-gnu
     timeout-minutes: 200
     steps:
-      - uses: actions/checkout@v3
-      - name: Set up QEMU
-        uses: docker/setup-qemu-action@v2
-      - name: Set up Docker Buildx
-        uses: docker/setup-buildx-action@v2
-      - name: Login to GitHub Container Registry
-        uses: docker/login-action@v2
-        with:
-          registry: ghcr.io
-          username: ${{ github.actor }}
-          password: ${{ secrets.GITHUB_TOKEN }}
-      - name: Build and push RPC
-        uses: docker/build-push-action@v4
-        with:
-          push: true
-          tags: ${{ env.REGISTRY_IMAGE }}:${{ github.ref_name }}, ${{ env.REGISTRY_IMAGE }}:latest
-          context: .
-          file: ./docker/rpc/Dockerfile
-          platforms: linux/amd64,linux/arm64
-          build-args: GITHUB_TOKEN=${{ secrets.GITHUB_TOKEN }}
-      - name: Build and Push Hive
-        uses: docker/build-push-action@v4
-        with:
-          push: true
-          tags: ghcr.io/${{ github.repository }}/hive:${{ github.ref_name }}, ghcr.io/${{ github.repository }}/hive:latest
-          context: .
-          file: ./docker/hive/Dockerfile
-          platforms: ${{ matrix.platform }}
-          build-args: |
-            GITHUB_TOKEN=${{ secrets.GITHUB_TOKEN }}
-            APIBARA_STARKNET_BIN_DIR=${{ matrix.STARKNET_BIN_DIR }}
-            APIBARA_SINK_BIN_DIR=${{ matrix.SINK_BIN_DIR }}
+      - uses: actions/checkout@v4
+      - name: Download artifacts
+        uses: actions/download-artifact@v4
+        with:
+          name: artifacts
+          path: /tmp
+      - run: mkdir -p ./lib/kakarot/build && mv /tmp/artifacts/contracts/* ./lib/kakarot/build
+      - run: mkdir ./artifacts && mv /tmp/artifacts/abi/* ./artifacts
+      - uses: actions/checkout@v4
+      - uses: dtolnay/rust-toolchain@stable
+      - uses: Swatinem/rust-cache@v2
+        with:
+          cache-on-failure: true
+      - uses: taiki-e/install-action@cross
+      - name: Log in to Docker
+        run: |
+          echo "${{ secrets.GITHUB_TOKEN }}" | docker login ghcr.io --username ${DOCKER_USERNAME} --password-stdin
```

Comment:
why do we need to manually connect to docker? i thought the action was doing it

Reply:
So in order to install `tonistiigi/binfmt`, which is a cross platform emulator, I need to run docker as privileged, and for this I need to be logged in.